File: 917-boats-to-save-people/boats-to-save-people.cpp
class Solution {
public:
    int numRescueBoats(vector<int>& people, int limit) {
        sort(people.begin(),people.end());

        int count=0;
        int l=0;
        int r=people.size()-1;

        while(l<=r){
            if(people[l]+people[r]<=limit){  
                l++;
            }
            count++;
            r--;
        }
        return count;
    }
};

File: 917-boats-to-save-people/boats-to-save-people.py
class Solution:
    def numRescueBoats(self, people: List[int], limit: int) -> int:
        people.sort()

        count=0

        l=0
        r=len(people)-1
        while l<=r:
            if people[l]+people[r]<=limit:
                l+=1
                r-=1
            else:
                r-=1
            count+=1
            
        return count